### PR TITLE
Fix setting pipe used in PAL_ProbeMemory as nonblocking

### PR DESCRIPTION
**Pull request by vtikoo**
Signed-off-by: Vikas Tikoo <vikasamar@gmail.com>

### CODE REVIEW

File: src/coreclr/pal/src/debug/debug.cpp [hunk 1]
```diff
@@ -742,15 +742,19 @@ PAL_ProbeMemory(
...
     if (pipe(fds) != 0)
     {
         ASSERT("pipe failed: errno is %d (%s)\n", errno, strerror(errno));
         return FALSE;
     }
 
-    fcntl(fds[0], O_NONBLOCK);
-    fcntl(fds[1], O_NONBLOCK);
+    flags = fcntl(fds[0], F_GETFL, 0);
+    fcntl(fds[0], F_SETFL, flags | O_NONBLOCK);
```
Comment (adamsitnik) [inline src/coreclr/pal/src/debug/debug.cpp:754]: an alternative would be using `pipe2`  but this would require an `#ifdef `

**Review by adamsitnik — APPROVED** [review]
@vtikoo have you hit any particular problem or just found a possible issue when analyzing the source code?

LGTM, thank you!

**Comment by vtikoo** [comment]
> @vtikoo have you hit any particular problem or just found a possible issue when analyzing the source code?

I hit this when running against a library OS our team is working on - [Mystikos](https://github.com/deislabs/mystikos).

File: src/coreclr/pal/src/debug/debug.cpp (hunk 1 above)
Comment (vtikoo) [reply src/coreclr/pal/src/debug/debug.cpp:754]: Do some targets not support `pipe2`?

Comment (adamsitnik) [reply src/coreclr/pal/src/debug/debug.cpp:754]: `pipe2` is Linux-specific and we support macOS and few other Unix-like OSese. The current solution is good, I am going to merge the PR.

**Pull request merged**